lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,26 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @warriors = building.warriors
+      @granary = building.granary
     end
 
     def call
-      raise NotImprementedYet
+      @warriors.present? ? @granary / daily_consumption : 0
+    end
+
+    private
+
+    def daily_consumption
+      sum = 10
+      @warriors.map do |warrior|
+        if warrior.type == 'Warriors::Samurai'
```

Comment:
Ładniej by to chyba wyglądało, jeśli Warrior miałby zaimplementowaną metodę/atrybut np. supply_burn_rate, który zwraca 1 jeśli warrior nie ma konia i 2 jeśli go ma, wtedy nie trzeba by myśleć nad takimi ifami w tym serwisie :)

Reply:
Dzięki za sprawdzenie i wszelkie uwagi :)